src/autocomplete.js

```diff
@@ -341,15 +346,26 @@ class Autocomplete {
 
             if (finished) {
                 // No results
-                if (!filtered.length)
+                if (!filtered.length && (!this.emptyMessage || this.autoShown))
                     return this.detach();
 
+                if (!filtered.length && this.emptyMessage && !this.autoShown) {
+                    var completionsForEmpty = [{
+                        caption: this.emptyMessage(prefix),
```

Comment:
Should this be shown  merely based on existence of emptyMessage method or also on it returning a non-empty string?
Something like the following:
```js
if (!filtered.length) {
    var emptyMessage = !this.autoShown && this.emptyMessage
    if ( typeof emptyMessage == "function")
          emptyMessage = this.emptyMessage(prefix);
     if (emptyMessage) { /*show completions code */}
     return this.detach();
}
```

Reply:
I've changed the `return.detach` tho. If we detach directly, no popover is shown.